contracts/loan_manager/src/contract.rs

```diff
@@ -378,6 +390,77 @@ impl LoanManager {
     pub fn get_interest_rate(e: Env, pool: Address) -> i128 {
         get_interest(e, pool)
     }
+
+    pub fn liquidate(e: Env, user: Address, borrower: Address, amount: i128) -> (i128, i128, i128) {
+        user.require_auth();
+
+        let Loan {
+            borrower,
+            borrowed_amount,
+            borrowed_from,
+            collateral_from,
+            collateral_amount,
+            health_factor,
+            unpaid_interest,
+        } = Self::get_loan(&e, borrower);
```

Comment:
Should we re-calculate the exact health factor again at this time?

Reply:
This would be a good fail safe as for now it's hard to say how efficiently the protocol keeps updating (e.g. how frequent usage we have). However, I am slightly worried about this causing the function to hit R-limits. I'll do some testing on how it works.